### PR TITLE
chore: code split for sidebar env split

### PR DESCRIPTION
**Pull request by ayushpahwa**
#### PR fixes following issue(s)
Fixes CE Issue (1428)

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

## Testing

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ayushpahwa** [comment]
Adding skipTestplan since this is a chore for code split. No change in user perception.

File: app/client/src/pages/common/datasourceAuth/index.tsx [hunk 1]
```diff
@@ -296,6 +300,7 @@ function DatasourceAuth({
           key={buttonType}
           kind="secondary"
           onClick={handleDatasourceTest}
+          showFilterComponent={showFilterComponent}
```
Comment (ChandanBalajiBP) [inline app/client/src/pages/common/datasourceAuth/index.tsx:303]: Isn't this a case of prop drilling? You are using passing the same param to many children, can we move this to store?

Comment (ayushpahwa) [reply app/client/src/pages/common/datasourceAuth/index.tsx:303]: image
Adding a image for more context. This is only going from parent to child one level down. Not sure if this needs to be moved. This variable is the same as the `pluginName` variable, also passed down from the parent to these components. In datasourceForm, this value is always false so doesn't matter their also.

**Review by ChandanBalajiBP — DISMISSED** [review]
LGTM

**Review by ChandanBalajiBP — APPROVED** [review]

**Pull request merged**